### PR TITLE
Engine: move the shader code inliner to the base ShaderProcessor class

### PR DESCRIPTION
**Pull request by Popov72**

### CODE REVIEW

File: src/Engines/Processors/shaderProcessor.ts [hunk 1]
```diff
@@ -306,7 +309,12 @@ export class ShaderProcessor {
             preparedSourceCode = options.processor.postProcessor(preparedSourceCode, defines, options.isFragment, options.processingContext, engine);
         }
 
-        return preparedSourceCode;
+        // Inline functions tagged with #define inline
+        let sci = new ShaderCodeInliner(preparedSourceCode);
```
Comment (bghgary) [inline src/Engines/Processors/shaderProcessor.ts:313]: nit: can be const
```suggestion
        const sci = new ShaderCodeInliner(preparedSourceCode);
```

**Review by bghgary — APPROVED** [review]

**Review by deltakosh — APPROVED** [review]

**Comment by sebavan** [comment]
@Popov72 and @bghgary  should this be merged ?

**Comment by Popov72** [comment]
That's ok for me.

**Comment by bghgary** [comment]
I approved, so yes.

**Comment by sebavan** [comment]
np was just wondering as it was staled I ll merge it now :-)

**Pull request merged**